docs/Core Mechanisms.md

```diff
@@ -12,69 +12,53 @@ NCA traffic between Controllers and Devices consists of three kinds of control m
 
-Once registered, a subscription remains in effect for the life of the control session, unless the Controller cancels it.
+Once registered, a subscription remains in effect for the life of the control session, unless the Controller cancels it or the Controller's connection fails.
 
-### Notification messages
+### Notification messages ###
 
 A notification message is essentially a callback command whose parameters depend on the type of event. However, unlike actual commands, a notification does not require a response.
 
 The parameters of notifications depend on the type of event. In general, event parameters are designed to give full detail about the event, so that Controllers receiving events can act on them immediately, without having to query the Device further.
 
-### Notification delivery
+### The PropertyChanged event ###
 
-The precise method by which notifications are delivered back to Controllers is a protocol issue, and out of scope of this document. However, for efficiency, the control model does identify two abstract ways of notification delivery. These are identified as _reliable_ and _fast_. **Reliable** delivery uses the same delivery method as Commands and Responses; **Fast** delivery uses lower-overhead, lower-reliability methods.
+The most common event is the **PropertyChanged** event.  **PropertyChanged**   is defined by the base class **ncaObject**, so it is inherited by every object. 
 
-**Fast** delivery may support multicasting; if so, Devices are allowed to multicast notifications for events to which there are multiple subscribers.
+**PropertyChanged** is the key mechanism for NCA's support of multiple Controllers without polling. By subscribing to changes in the properties it cares about, a Controller can stay in sync with the device, even when those properties are being changed by other controllers. 
 
-Controllers specify the type of delivery separately for each subscription. This allows them to elect **Reliable** delivery for critical events such as error alerts, and **Fast** delivery for noncritical events such as signal level sensor updates.
+## Controller hierarchies ##
 
-The control model says nothing about how **Reliable** and **Fast** delivery schemes are implemented. Their implementation will vary from protocol to protocol. In some cases, the two may be the same. **Fast** may or may not be able to multicast. The control model only defines the abstraction, not the realization.
+Some NCA-controlled media networks may contain thousands of devices. In such cases, it will not be practical for a single central Controller to maintain a separate control session with each device. To address this problem, indirect control using a hierarchy of Controllers may be implemented, with successive levels aggregating control functionality in ways appropriate to the applications.  NCA's core mechanisms are sufficient to support such configurations, but the exact implementation details are out of this specification's scope.
```

Comment:
```suggestion
Some NCA-controlled media networks may contain thousands of devices. In such cases, it will not be practical for a single central Controller to maintain a separate control session with each device. To address this problem, indirect control using a hierarchy of Controllers can be implemented, with successive levels aggregating control functionality in ways appropriate to the applications.  NCA's core mechanisms are sufficient to support such configurations, but the exact implementation details are out of this specification's scope.
```